server/chat-plugins/abuse-monitor.ts

```diff
@@ -42,16 +42,31 @@ const REPORT_NAMECOLORS: {[k: string]: string} = {
 	other: '', // black - empty since handled by dark mode
 };
 
+export let migrated = global.Chat?.oldPlugins['abuse-monitor']?.migrated || false;
+
 export const cache: {
 	[roomid: string]: {
 		users: Record<string, number>,
 		// can be a string and not string[] if it was added to the object before this patch was done.
 		// todo: move this to just ID[]
 		staffNotified?: ID | ID[],
 		claimed?: ID,
-		recommended?: {type: string, reason: string},
+		recommended?: Record<string, {type: string, reason: string}>,
 	},
-} = global.Chat?.oldPlugins['abuse-monitor']?.cache || {};
+} = (() => {
+	const plugin = global.Chat?.oldPlugins['abuse-monitor'];
+	if (!plugin?.cache) return {};
+	if (plugin.migrated) return plugin.cache;
+	for (const k in plugin.cache) {
+		const cur = plugin.cache[k];
+		if (typeof cur.recommended?.type === 'string') { // would be object if it was the new entry
+			// we cannot feasibly determine who it was (but it __is__ logged in <<abuselog>>, so staff can)
+			delete cur.recommended;
+		}
```

Comment:
Should we really discard data about recommendations just because we don't know which user it was for?

Reply:
Not like there's a better way to do it. Plus, it's logged to abuselog anyway, so staff can still see it.

Reply:
You could save it under something like `an unknown user`, but it's probably fine as-is.